spec/controllers/spree/graphql_controller_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+RSpec.describe Spree::GraphqlController do
+  describe '#graphql' do
+    let(:action) { post '/graphql', params: params, headers: headers }
+    let(:params) { { } }
+    let(:headers) { { } }
+
+    context 'when the authentication is required' do
+      before do
+        Spree::GraphQL::Config.set(requires_authentication: false)
+        action
```

Comment:
If you make the `post` request in the `subject`, You don't have to make this `action` call.

Reply:
do you propose the following?

```ruby
before do
  Spree::GraphQL::Config.set(requires_authentication: false)
end

subject do
  action
  response
end
```

Reply:
No, this is my propose:
```
before do
  Spree::GraphQL::Config.set(requires_authentication: false)
end

subject { post '/graphql', params: params, headers: headers }
```
What do you think? :smile:

Reply:
@EduardoGHdez I can't do it, I need the response object in the tests, otherwise I can't use `have_http_status`

Reply:
You will receive the `respone` :smile:

Reply:
yes but I can't use the implicit subject if the response isn't the subject

Reply:
Yes, you can :thinking: 
I already test this approach, but it was just a suggesion :smile: 
Or is a bad practice? :astonished: 
![image](https://user-images.githubusercontent.com/39539196/55018595-cfc6cd80-4fb8-11e9-9902-56472e0141e0.png)

Reply:
BTW, thanks for attend my comments :smile:

Reply:
@EduardoGHdez it's my pleasure! but you are proposal doesn't work here:

```ruby
      before do
        Spree::GraphQL::Config.set(requires_authentication: false)
        # action
      end

      # subject { action }
      subject { response }
```

![Schermata del 2019-03-26 18-29-29](https://user-images.githubusercontent.com/283320/55019407-1d612b80-4ff5-11e9-94ae-78bdae26e09c.png)

Reply:
That's because you didn't set the `post` in the subject
Try with this :smile: 
```
before do
  Spree::GraphQL::Config.set(requires_authentication: false)
end

subject { post '/graphql', params: params, headers: headers }
```

Reply:
Or 
`subject { action } `

Reply:
@EduardoGHdez still the same error! try to clone my fork, checkout this branch and make it work! But now I'm confused, because I tried to run specs within https://github.com/EduardoGHdez/agenda ... and it works there! Could it be some RSpec configuration difference? I don't know :flushed:

Reply:
@EduardoGHdez I used `subject { action }` before, it was a typo

Reply:
Nice to see the collbaration!  Good work!

Reply:
i guess you are right, It could be an RSpec configuration.
I didn't find another explanation.
But, again, thanks for attend  my comments :smile:

Reply:
@EduardoGHdez fixed!! look at this: https://github.com/boomerdigital/solidus_graphql_api/pull/36/commits/27debd98e62704f8adfb80f24abfec71f1253250

Reply:
Wow! :tada: 
I already saw your solution @mdesantis. 
Thank you too! :smile: